pkg/netflow/config/config.go

```diff
@@ -0,0 +1,83 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2020-present Datadog, Inc.
+
+package config
+
+import (
+	"fmt"
+
+	coreconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/netflow/common"
+)
+
+// NetflowConfig contains configuration for NetFlow collector.
+type NetflowConfig struct {
+	Listeners               []ListenerConfig `mapstructure:"listeners"`
+	StopTimeout             int              `mapstructure:"stop_timeout"`
+	AggregatorBufferSize    int              `mapstructure:"aggregator_buffer_size"`
+	AggregatorFlushInterval int              `mapstructure:"aggregator_flush_interval"`
+	LogPayloads             bool             `mapstructure:"log_payloads"`
+}
+
+// ListenerConfig contains configuration for a single flow listener
+type ListenerConfig struct {
+	FlowType  common.FlowType `mapstructure:"flow_type"`
+	Port      uint16          `mapstructure:"port"`
+	BindHost  string          `mapstructure:"bind_host"`
+	Workers   int             `mapstructure:"workers"`
+	Namespace string          `mapstructure:"namespace"`
+}
+
+// ReadConfig builds and returns configuration from Agent configuration.
+func ReadConfig() (*NetflowConfig, error) {
+	var mainConfig NetflowConfig
+
+	err := coreconfig.Datadog.UnmarshalKey("network_devices.netflow", &mainConfig)
+	if err != nil {
+		return nil, err
+	}
+	for i := range mainConfig.Listeners {
+		listenerConfig := &mainConfig.Listeners[i]
+
+		flowType, err := common.GetFlowTypeByName(listenerConfig.FlowType)
+		if err != nil {
+			return nil, fmt.Errorf("the provided flow type `%s` is not valid (valid flow types: %v)", listenerConfig.FlowType, common.GetAllFlowTypes())
+		}
+
+		if listenerConfig.Port == 0 {
+			listenerConfig.Port = flowType.DefaultPort()
+			if listenerConfig.Port == 0 {
+				return nil, fmt.Errorf("no default port found for `%s`, a valid port must be set", listenerConfig.FlowType)
+			}
+		}
+		if listenerConfig.BindHost == "" {
```

Comment:
small blocker: I assume flows will usually be sent by an "exporter" device and be sent over the network to the agent. Likely the default GetBindHost value of 127.0.0.1 is not a good default.

Reply:
make sense, updated here fa4cd359f